tests/Network/ConnectionPoolTest.cpp

```diff
@@ -365,107 +434,235 @@ TEST_F(NetworkConnectionPoolTest, checking_expiration_multiple_endpints) {
 
   // 21ms > 2 * 10ms
   std::this_thread::sleep_for(std::chrono::milliseconds(21));
 
   // expires the connections
   pool.pruneConnections();
   ASSERT_EQ(pool.numOpenConnections(), 0);
+  EXPECT_EQ(extractCurrentMetric(), 0ull);
 
   pool.drainConnections();
+  ASSERT_EQ(pool.numOpenConnections(), 0);
+  EXPECT_EQ(extractCurrentMetric(), 0ull);
 
   {
     auto conn1 = pool.leaseConnection("tcp://example.org:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 1);
+    EXPECT_EQ(extractCurrentMetric(), 1ull);
 
     auto conn2 = pool.leaseConnection("tcp://example.org:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 2);
+    EXPECT_EQ(extractCurrentMetric(), 2ull);
 
     auto conn3 = pool.leaseConnection("tcp://example.org:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 3);
+    EXPECT_EQ(extractCurrentMetric(), 3ull);
   }
   {
     auto conn4 = pool.leaseConnection("tcp://example.com:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 4);
+    EXPECT_EQ(extractCurrentMetric(), 4ull);
 
     auto conn5 = pool.leaseConnection("tcp://example.com:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 5);
+    EXPECT_EQ(extractCurrentMetric(), 5ull);
 
     // 21ms > 2 * 10ms
     std::this_thread::sleep_for(std::chrono::milliseconds(21));
 
     // expires the connections
     pool.pruneConnections();
     ASSERT_EQ(pool.numOpenConnections(), 2);
+    EXPECT_EQ(extractCurrentMetric(), 2ull);
   }
 
   pool.drainConnections();
+  ASSERT_EQ(pool.numOpenConnections(), 0);
+  EXPECT_EQ(extractCurrentMetric(), 0ull);
 
   {
     auto conn1 = pool.leaseConnection("tcp://example.org:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 1);
+    EXPECT_EQ(extractCurrentMetric(), 1ull);
 
     auto conn2 = pool.leaseConnection("tcp://example.com:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 2);
+    EXPECT_EQ(extractCurrentMetric(), 2ull);
 
     auto conn3 = pool.leaseConnection("tcp://example.net:80", isFromPool);
     ASSERT_EQ(pool.numOpenConnections(), 3);
+    EXPECT_EQ(extractCurrentMetric(), 3ull);
   }
   ASSERT_EQ(pool.numOpenConnections(), 3);
+  EXPECT_EQ(extractCurrentMetric(), 3ull);
 
   // 21ms > 2 * 10ms
   std::this_thread::sleep_for(std::chrono::milliseconds(21));
 
   // expires the connections
   pool.pruneConnections();
   ASSERT_EQ(pool.numOpenConnections(), 0);
+  EXPECT_EQ(extractCurrentMetric(), 0ull);
 
   pool.drainConnections();
+  ASSERT_EQ(pool.numOpenConnections(), 0);
+  EXPECT_EQ(extractCurrentMetric(), 0ull);
+}
+
+TEST_F(NetworkConnectionPoolTest, test_cancel_endpoint_all) {
+  ConnectionPool::Config config(server.getFeature<metrics::MetricsFeature>());
+  config.numIOThreads = 1;
+  config.maxOpenConnections = 2;
+  config.idleConnectionMilli = 10;  // extra small for testing
+  config.verifyHosts = false;
+  config.protocol = fuerte::ProtocolType::Http;
+
+  ConnectionPool pool(config);
+  std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
Just wondering why an extra sleep at the beginning of a test should be necessary :thinking: